weaviate/weaviate_client.go

```diff
@@ -41,12 +41,16 @@ type Config struct {
 func NewConfig(host string, scheme string, authConfig auth.Config, headers map[string]string) (*Config, error) {
```

Comment:
This function should also work when headers is nil

Note a test for this would be great, you could do [a table driven test](https://yourbasic.org/golang/table-driven-unit-test/) that uses:
a) nil
b) empty map
c) map with preexisting values

as value for `headers`